src/js/review/ArrayField.jsx

```diff
@@ -109,6 +109,10 @@ class ArrayField extends React.Component {
     });
   }
 
+  handleNewItemTitleMounted = element => {
+    element.focus();
```

Comment:
Are you sure it's safe to do this? At minimum, you're not calling this when the element is mounted, but when the ref is set, so the name is somewhat deceptive. The docs say [`ref` updates happen before `componentDidMount` or `componentDidUpdate` lifecycle methods](https://reactjs.org/docs/refs-and-the-dom.html#adding-a-ref-to-a-dom-element).

The main reason why this might matter is because in many or most browsers you can't set focus to an element unless it's currently in the DOM. If React is constructing a component off-DOM the attempt to set focus will fail.

Reply:
Good call- I added a truthy check. My understanding is that the `ref` callback is tied to the DOM manipulation so it wouldn't be called until the element is created an exists in the object model.